### PR TITLE
Add FTM Fantom Opera mainnet 250

### PR DESCRIPTION
**Pull request by Kharabet**
# Add New Chain 250

Thanks for your pull request to add a new support in Sourcify.

If you haven't done so, please follow the instructions on [how to request chain support](https://docs.sourcify.dev/docs/chain-support/) in docs.

Please check the following items before submitting your pull request.

## Checklist

- [x] The branch is named as `add-chain-<chainId>`.
- [x] I haven't modified the [chains.json](../../src/chains.json) file directly.
- [x] In [sourcify-chains.ts](../../src/sourcify-chains.ts) file
  - [x] I've set `supported: true`.
  - [x] I've set `monitored: false`.
  - [x] I haven't added an `rpc` field but the one in [chains.json](../../src/chains.json) is used (if not, please explain why).
- [x] I've added a test in [chain-tests.js](../../test/chains/chains-test.js) file.
- [x] `test-new-chain` test in Circle CI is passing.

### CODE REVIEW

**Comment by Kharabet** [comment]
It says it can't fetch GSN/Context.sol. But here it is https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v2.5.0/contracts/GSN/Context.sol

image

**Comment by kuzdogan** [comment]
It can't fetch it because there's no GitHub url under the `url`s of that file. There's only the `bzzr` hash. 

Can you also add those files alongside the test files as well as add them in the `verifyContract`'s sources array?

**Comment by Kharabet** [comment]
@kuzdogan any reason it can't see FUSDToken.sol now?

File: src/sourcify-chains.ts [hunk 1]
```diff
@@ -957,10 +957,15 @@ const sourcifyChainsExtensions: SourcifyChainsExtensionsObject = {
     // Edgeware EdgeEVM Mainnet
     supported: true,
     monitored: false,
-    contractFetchAddress:
```
Comment (kuzdogan) [inline src/sourcify-chains.ts:960]: Is this removal on purpose?

**Comment by Kharabet** [comment]
> Hmm from what I can tell the `FUSDToken.sol` file provided has a different hash (`0x430f6a7...`) than in the metadata (`0xdffa824...`). Are you sure these are the original files? If this contract is in a repository somewhere, I can also have a look. Alternatively we could've gotten an already verified contract from https://ftmscan.com/ using the "Import from Etherscan" but I see it's down.

Ah, I see. in sol file I changed imports so `@opnezeppelin` files are taken not from github, but locally

I took fantom's stablecoin as a contract for the test https://github.com/Fantom-foundation/Fantom-Fusd

**Comment by kuzdogan** [comment]
Ok I can have a look at this, if you can enable maintainer access to this PR https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

File: src/sourcify-chains.ts (hunk 1 above)
Comment (Kharabet) [reply src/sourcify-chains.ts:960]: nope, my fault. as well as some formatting. I will disable my VS code formatter for the workspace and push the changes. Thanks!

**Comment by Kharabet** [comment]
> Ok I can have a look at this, if you can enable maintainer access to this PR https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

✅
image

**Comment by Kharabet** [comment]
> Ok I can have a look at this, if you can enable maintainer access to this PR https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

one sec, I have an idea what to do. Let me try

**Comment by Kharabet** [comment]
Hey @kuzdogan, I changed the contract for test: deployed Storage contract on Fantom chain https://ftmscan.com/address/0xc47856bEBCcc2BBB23E7a5E1Ba8bB4Fffa5C5476#code

**Comment by kuzdogan** [comment]
Thanks a lot for deploying. Good to go now 👍
